NetCasbin/Model/Policy.cs

```diff
@@ -73,29 +83,37 @@ public List<List<string>> GetFilteredPolicy(string sec, string ptype, int fieldI
 
         public bool HasPolicy(string sec, string ptype, List<string> rule)
         {
-            return Model[sec][ptype]!=null && Model[sec][ptype].Policy.Any(x => Utility.ArrayEquals(rule, x));
+            return Model[sec][ptype] != null && Model[sec][ptype].PolicyStringSet.Contains(Utility.ArrayToString(rule));
         }
 
         public bool AddPolicy(string sec, string ptype, List<string> rule)
         {
-            if (!HasPolicy(sec, ptype, rule))
+            if (HasPolicy(sec, ptype, rule))
             {
-                Model[sec][ptype].Policy.Add(rule);
-                return true;
+                return false;
             }
-            return false;
+
+            Assertion assertion = Model[sec][ptype];
+            assertion.Policy.Add(rule);
```

Comment:
Can we merge these 2 steps into 1? We can compare 2 List, why change to string

Reply:
If changed `List<List<string>>` to `Hash<Rule>` (the `Rule` inherit `List<string>` and overwrite the `Equals` and `GetHashCode`), It will change almost all the public API and can not gain more performance. It will cost much time when `Add` the list because the `Equal` or `Add` has  O(n) complexity.

Reply:
If we have a lot of policies in memory, will this increase memory usage?

Reply:
You are right,  But it will not much increase memory and to file adapter it can be like a cache.
We can try to make it better, I think we can convert it to an array when using it.

Reply:
the `assertion.PolicyStringSet` proptry does not need to be public. 
1. add  `assertion.AddPolicy(rule)` method in `Assertion Class` replace  these 2 steps:
 ```c#
assertion.Policy.Add(rule);
assertion.PolicyStringSet.Add(Utility.ArrayToString(rule));
```

2. add `assertion.Contains(rule)` mehod  in `Assertion Class` replace ` Model[sec][ptype].PolicyStringSet.Contains(Utility.ArrayToString(rule))`

3. add `assertion.RemovePolicyAt(i)` method  replace  these 2 steps:
```c#
assertion.Policy.RemoveAt(i);
assertion.PolicyStringSet.Remove(Utility.ArrayToString(rule));
```

Reply:
Ok, The `assertion.PolicyStringSet` is Internal now, I will add them at next commit.